### ISSUE TITLE
Walk command launching NPCs towards 0,0

### ISSUE THREAD

**Issue by teateatea** [DenizenScript/Denizen-For-Bukkit#1160]
Hello!

I'm using the walk command to create patrolling NPCs.

Haste script here: http://mcmonkey.org/haste/20228

In this script, the NPC will randomly select one of three patrol points and walk to it. A single NPC has no issue patrolling for extended periods of time.

Unfortunately, only one or two NPCs can run this script at a time.

At 3+ NPCs, eventually one of them will zoom away. He will still be facing his destination and animating his walk, but the NPC is physically sliding away.

- He is stopped by walls, so it does not appear to be a teleport.
- There does not appear to be any pattern, it does not happen consistently. Random NPC (sometimes several at once), random origin, random destination.
- The distance between points doesn't seem to matter either, although for the purposes of testing I've mostly used ~10 blocks between each point.
- The only consistent factor is that they always shoot towards 0,0.
- They are still trying to get to walk to their destination after their zoom. You'll see them walking back out of the forest once they get stopped by the trees.

GIF 1: https://gyazo.com/872f72226ca3ab333373491d46551fd0
(Note the topmost-Steve sliding into the forest, while facing yellow. They never look like they're _trying_ to go to 0,0)
GIF 2: https://gyazo.com/2ea9085e3807e5e3167cf78536d44948
(The issue is exacerbated by more NPCs)
GIF 2: https://gyazo.com/1928996d439363c1d68421d003d9b94a
(Note the bottom Steve getting yanked mid-walk)

Is this a bug on your end?

Am I setting up my script wrong? Is there a way for me to solve this, or work around this?

Thank you for you time,
Tea

**Comment by teateatea** [comment]
The issue appears to be in the radius argument of the walk command.

Removing "radius:5" has largely resolved the "zooming" issue, and putting it back immediately brings the "zooming" right back. No other changes were made to the script above.

A huge thank you to Talamar1 for his help and patience with me lol.

**Comment by Talamar1** [comment]
@fullwall is this an issue with the Flocker in Citizen not computing the target location correctly when the destination is outside the radius?

**Comment by teateatea** [comment]
Talamar1 gave me a new citizens jar from fullwall, and I've tried it out with the above script.

Technically, it did solve the original zooming issue, but some other issues have popped up instead (on that jar).

NPCs have trouble actually arriving to their destination. Very often, they reach the destination, but do a spinning dance. They don't stop spinning once they do, and the script never gives them a new destination, presumably because they never arrive at the original one.
GIF 4: https://gyazo.com/18ff098d074436315343c1a8b034729b
These NPCs are spinning! And they can't stop!

Some NPCs behave even more erratically. 2 out of 5 did not spin, but instead appeared to be disappearing, and then teleporting back to their original location from below the ground. The initial disappear gives no hints as to their location, but as they return you can occasionally see their nameplate underground just before they reappear. The Alex on the left side of GIF 4 is demonstrating this behaviour. She continued to blink with no obvious intent to actually walk to her destination.

Furthermore, pausing waypoints at any point after they've been given a walk command seems to cause serious issues. In my script, right-clicking the NPC while it's patrolling issues a pause waypoints command, but with this jar, they all disappeared on click. I did not know where they went, so I did a /npc tp command to find out. This immediately kicked me with a "Invalid move packet received", and further log-ins indicated that I was trying to log in at <[world]256.295, NaN, -52,440> (and again, kicked for Internal Exception).

**Comment by teateatea** [comment]
Okay! Just received another citizens jar from fullwall, tested in the same way as above.

All of the issues from my last post appear to have been resolved.

The original zooming issue appears to have been resolved.

I would note two further bugs in this version:
1) The radius argument in the walk command no longer causes obvious issues, but doesn't seem to actually do anything. I tried radius:5 and radius:10, and both resulted in the NPC finding his way to within 1 block of the destination.
2) For some reason the radius argument results in decreased speed in the NPCs walk. They appear to slow down to about 50-60% of their usual walk speed. Removing the radius argument returns the NPCs to their usual walk speed.

Improvements nonetheless!
